### PR TITLE
Parallel sort

### PR DESCRIPTION
**Pull request by thilinarmtb**
## TODO
- [x] Refactor tests
- [x] Test at scale

### CODE REVIEW

File: src/sort/binsort.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      if(val<=end) proc[index++]=id;
+      else break;
+    }
+    id++;
+  }while(id<np && index<size);
+  for(; index<size; index++)
+    proc[index]=np-1;
+}
+
+int exaBinSort(sort_data data,struct comm *c)
```
Comment (thilinarmtb) [inline src/sort/binsort.c]: Remove 'exa'.

File: src/sort/local.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#if 0 //FIXME
+    case gs_float:
+        gslib_sortp_float (buf,keep,(float *)((char*)ptr+off),
+            nunits,usize);
+      break;
+#endif
+    case gs_long:
+      gslib_sortp_ull(buf,keep,(ulong *)((char*)ptr+off),nunits,usize);
+      break;
+    case gs_int:
```
Comment (thilinarmtb) [inline src/sort/local.c:22]: Using `gs_long` and `gs_int` temporarily.

File: src/sort/sort.c [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    gs_dom type)
+{
+  char* v=(char*)a->ptr+i*usize+offset;
+  double data;
+
+  switch(type){
+    case gs_int:
+      data=*((uint*)v);
+      break;
+    case gs_long:
```
Comment (thilinarmtb) [inline src/sort/sort.c:62]: Using `gs_long` and `gs_int` temporarily.

File: src/sort/sort.h [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+void get_extrema(void *extrema_,sort_data data,uint field,struct comm *c);
+int  set_dest(uint *proc,uint np,ulong start,uint size,ulong nelem);
+int  load_balance(struct array *a,size_t size,struct comm *c,
+  struct crystal *cr);
+int  sort_local(sort_data data);
+int  sort_private(sort_data data,struct comm *c);
+//
+// exaBinSort
+//
+int exaBinSort(sort_data data,struct comm *c);
```
Comment (thilinarmtb) [inline src/sort/sort.h]: Rename function names and enums starting with `exa`.

**Comment by stgeke** [comment]
Replaced by #20

**Pull request closed without merging**